drivers/can/can_sam0.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2022 Vestas Wind Systems A/S
+ * Copyright (c) 2021 Alexander Wachter
+ * Copyright (c) 2022 Kamil Serwus
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/can.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include "can_mcan.h"
+
+LOG_MODULE_REGISTER(can_sam0, CONFIG_CAN_LOG_LEVEL);
+
+#define DT_DRV_COMPAT atmel_sam0_can
+
+struct can_sam0_config {
+	void (*config_irq)(void);
+	const struct pinctrl_dev_config *pcfg;
+	volatile uint32_t *mclk;
+	uint32_t mclk_mask;
+	uint16_t gclk_core_id;
+	int divider;
+};
+
+struct can_sam0_data {
+	struct can_mcan_msg_sram msg_ram;
+};
+
+void can_sam0_line_x_isr(const struct device *dev) {
+	can_mcan_line_0_isr(dev);
+	can_mcan_line_1_isr(dev);
+}
```

Comment:
Does the SAM0 series only use one M_CAN interrupt line?

Reply:
Yes. SAM21 have only one interrupt line. But I saw that can_mcan is expected 2 lines -> I fire 2 ISRs